src/transformers/models/cvt/modeling_cvt.py

```diff
@@ -0,0 +1,750 @@
+            proj = None
+        else:
+            raise ValueError("Unknown method ({})".format(method))
+
+        return proj
+
+    def forward_conv(self, x, h, w):
+        if self.with_cls_token:
+            cls_token, x = torch.split(x, [1, h * w], 1)
+
+        x = rearrange(x, "b (h w) c -> b c h w", h=h, w=w)
+
+        if self.conv_proj_q is not None:
+            q = self.conv_proj_q(x)
+        else:
+            q = rearrange(x, "b c h w -> b (h w) c")
+
+        if self.conv_proj_k is not None:
+            k = self.conv_proj_k(x)
+        else:
+            k = rearrange(x, "b c h w -> b (h w) c")
+
+        if self.conv_proj_v is not None:
+            v = self.conv_proj_v(x)
+        else:
+            v = rearrange(x, "b c h w -> b (h w) c")
+
+        if self.with_cls_token:
+            q = torch.cat((cls_token, q), dim=1)
+            k = torch.cat((cls_token, k), dim=1)
+            v = torch.cat((cls_token, v), dim=1)
+
+        return q, k, v
+
+    def forward(self, x, h, w):
+        if self.conv_proj_q is not None or self.conv_proj_k is not None or self.conv_proj_v is not None:
+            q, k, v = self.forward_conv(x, h, w)
+
+        q = rearrange(self.proj_q(q), "b t (h d) -> b h t d", h=self.num_heads)
+        k = rearrange(self.proj_k(k), "b t (h d) -> b h t d", h=self.num_heads)
+        v = rearrange(self.proj_v(v), "b t (h d) -> b h t d", h=self.num_heads)
+
+        attn_score = torch.einsum("bhlk,bhtk->bhlt", [q, k]) * self.scale
+        attn = F.softmax(attn_score, dim=-1)
+        attn = self.attn_drop(attn)
+
+        x = torch.einsum("bhlt,bhtv->bhlv", [attn, v])
+        x = rearrange(x, "b h t d -> b t (h d)")
+
+        x = self.proj(x)
+        x = self.proj_drop(x)
+
+        return x, attn
+
+
+class CvTLayer(nn.Module):
+    def __init__(
+        self,
+        dim_in,
+        dim_out,
+        num_heads,
+        mlp_ratio=4.0,
+        qkv_bias=False,
+        drop=0.0,
+        attn_drop=0.0,
+        drop_path=0.0,
+        act_layer=nn.GELU,
+        norm_layer=nn.LayerNorm,
+        **kwargs
+    ):
+        super().__init__()
+
+        self.with_cls_token = kwargs["with_cls_token"]
+
+        self.norm1 = norm_layer(dim_in)
+        self.attn = CvTAttention(dim_in, dim_out, num_heads, qkv_bias, attn_drop, drop, **kwargs)
+
+        self.drop_path = DropPath(drop_path) if drop_path > 0.0 else nn.Identity()
+        self.norm2 = norm_layer(dim_out)
+
+        dim_mlp_hidden = int(dim_out * mlp_ratio)
+        self.mlp = CvTOutput(in_features=dim_out, hidden_features=dim_mlp_hidden, act_layer=act_layer, drop=drop)
+
+    def forward(self, x, h, w):
+        res = x
+
+        x = self.norm1(x)
+        attn, all_attn = self.attn(x, h, w)
+        x = res + self.drop_path(attn)
+        x = x + self.drop_path(self.mlp(self.norm2(x)))
+        return x, all_attn
+
+
+class CvTEncoder(nn.Module):
+    """Vision Transformer with support for patch or hybrid CNN input stage"""
```

Comment:
```suggestion
    """Vision Transformer with support for patch or hybrid CNN input stage"""
```
Don't think this class supports the hybrid CNN input stage.